### ISSUE TITLE
Run script only when there is a switch.

### ISSUE THREAD

**Issue by bigplayer-ai** [t1m0thyj/WDD-scripts#36]
Hey,
Hour can I run a script only once when there is switch.
For example when night mode is activated/dark theme is activated.
And not every time dark theme is on. (There is a big difference).

**Comment by t1m0thyj** [comment]
Currently scripts are invoked every time the value of any of the arguments changes (`daySegment2`, `daySegment4`, `nightMode`, or `imagePath`).

If you want your script to only perform an action when `nightMode` changes, you could do one of the following:
* Check if machine is already in the desired state before performing an action. For example in the mouse theme switch script, if `nightMode` is true, and the registry value "W11 Cursors Dark HD v2.2 by Jepri Creations" is already set, then the script can exit without performing any actions.
* Store variables to disk to reference them the next time the script is run - you can do this with the PowerShell cmdlets `Export-Clixml` and `Import-Clixml`.

**Comment by bigplayer-ai** [comment]
> 



> * Store variables to disk to reference them the next time the script is run - you can do this with the PowerShell cmdlets `Export-Clixml` and `Import-Clixml`.

thanks for this tip, I found a temporary not so cool solution for windows not respecting theme changes. I will pull request later on today.

**Comment by bigplayer-ai** [comment]
I want to set the saved night-value global variable to the night value of WDD on system startup.
How can I make another script to run only on system startup and get the details from WDD?

**Comment by bigplayer-ai** [comment]
check my new pull request #31  any tips would be really appreciated (I am new to coding).
